pregnancy-postpone.lua

```diff
@@ -0,0 +1,328 @@
+local help = [====[
+pregnancy-postpone
+=============
+Delays childbirths by a week or an user-definable amount of time (in ticks).
+Used as a workaround for a crash involving werebeasts giving birth. (https://www.bay12games.com/dwarves/mantisbt/view.php?id=11549)
+Can also be used simply to gather information about pregnancies in the world/playable area.
+Can act on: a selected unit, citizens on the map, non-citizens on the map, or worldwide.
+
+Examples
+=============
+"pregnancy-postpone --info-only --citizens --non-citizens"
+    Prints pregnancy information about all creatures in the playable area.
+
+"pregnancy-postpone --world --ticks 1200"
+    Postpones historical figures' childbirths worldwide by one day. (including those on your map)
+
+"pregnancy-postpone -iu"
+    Prints pregnancy information about the selected unit.
+
+"pregnancy-postpone -u"
+    Postpones selected unit's childbirth by a week. (the default amount of time)
+
+Options
+=============
+
+"-i" "--info-only"
+    Print pregnancy information without changing anything.
+
+"-s" "--silent"
+    Do not print pregnancy info.
+
+"-u" "--unit"
+    Acts only on the selected unit.
+
+"-c" "--citizens"
+    Acts on fort-controlled citizens in the playable area. (excludes animals)
+
+"-x" "--non-citizens"
+    Acts on all non-citizen units in the playable area, like animals and guests. (excludes citizens)
+
+"-w" "--world"
+    Act on the whole world. (INCLUDING your map, for now)
+
+"-t <number>" "--ticks <number>""
+    Postpone births by this many ticks.
+    If unspecified, the default value is 8400. (1 week)
+    in-game-time conversion table:
+      1 hour  = 50 ticks
+      1 day   = 1200 ticks
+      1 week  = 8400 ticks
+      1 month = 33600 ticks
+      1 year  = 403200 ticks
+
+Notes
+=============
+Short optional arguments can be chained together!
+   this command: "postpone-pregnancy --silent --citizens --ticks 2400"
+is identical to: "postpone-pregnancy -sct2400"
+Just make sure that the "-t <number>" argument comes last, if you want to use it.
+
+For best results, be sure that the game is paused while running the script!
+]====]
+--[[ Credits:
+---------------
+made by TaxiService during a secretive mood.
+  super duper turbo thanks to:
+<3 ab9rf - for hunting down the cause of my map's crash with a debugger, setting me off on this coding journey.
+<3 sdegrace - for mentioning how to use the lua console and "~" to print all. Invaluable tools for testing!
+<3 myk002, Lethosor, nuvu (vallode), ab9rf (again) - for always pointing me in the right direction and putting up with me and my questions!
+<3 all the giants from the DFHack community, on whose shoulders i stood on!
+
+notes for function grafters:
+  Whenever i refer to a "dude", it means a "unit OR historical_figure"
+  ...otherwise, i tried to go for "readable" over "compact". hopefully it'll be understandable enough!
+--]]
+local argparse = require 'argparse'
+local args = {

```

Comment:
in almost all scripts, this would be called `opts`, not `args` (which is usually used for incoming arguments, e.g. `local args = {...}`)

Reply:
ah crap. well, it's an easy change to make, at least!